pkg/logger/dapr_logger.go

```diff
@@ -0,0 +1,185 @@
+package logger
+
+import (
+	"os"
+	"time"
+
+	"github.com/dapr/dapr/pkg/version"
+	"github.com/dapr/dapr/utils"
+	"github.com/sirupsen/logrus"
+)
+
+// daprLogger is the implemention for logrus
+type daprLogger struct {
+	// name is the name of logger that is published to log as a scope
+	name string
+	// jsonFormatEnabled is a flag to turn on JSON formatted log
+	jsonFormatEnabled bool
+	// loger is the instance of logrus logger
+	logger *logrus.Entry
+}
+
+func newDaprLogger(name string) *daprLogger {
+	newLogger := logrus.New()
+
+	dl := &daprLogger{
+		name:              name,
+		jsonFormatEnabled: defaultJSONOutput,
+		logger: newLogger.WithFields(logrus.Fields{
+			logFieldScope: name,
+			logFieldType:  LogTypeLog,
+		}),
+	}
+
+	dl.EnableJSONOutput(defaultJSONOutput)
+
+	return dl
+}
+
+// EnableJSONOutput enables JSON formatted output log
+func (l *daprLogger) EnableJSONOutput(enabled bool) error {
+	var formatter logrus.Formatter
+
+	l.jsonFormatEnabled = enabled
+	fieldMap := logrus.FieldMap{
+		// If time field name is conflicted, logrus adds "fields." prefix.
+		// So rename to unused field @time to avoid the confliction.
+		logrus.FieldKeyTime:  "@time",
+		logrus.FieldKeyLevel: logFieldLevel,
+		logrus.FieldKeyMsg:   logFieldMessage,
+	}
+
+	l.logger.Data = logrus.Fields{
+		logFieldScope: l.logger.Data[logFieldScope],
+		logFieldType:  LogTypeLog,
+	}
+
+	if enabled {
+		hostname, _ := os.Hostname()
+		l.logger.Data[logFieldInstance] = hostname
+		l.logger.Data[logFieldDaprVer] = version.Version()
+
+		formatter = &logrus.JSONFormatter{
+			DisableTimestamp: true,
+			FieldMap:         fieldMap,
+		}
+	} else {
+		formatter = &logrus.TextFormatter{
+			DisableTimestamp: true,
+			FieldMap:         fieldMap,
+		}
+	}
+
+	l.logger.Logger.SetFormatter(formatter)
+
+	return nil
+}
+
+// SetDaprID sets dapr_id field in log. Default value is empty string
+func (l *daprLogger) SetDaprID(id string) {
```

Comment:
Is this Dapr Id or App Id? We had feedback asking to be consistent and use App Id only.

Reply:
Ah ok. I will change to app_id then

Reply:
fixed.